### PR TITLE
elasticsearch reindex management prework

### PR DESCRIPTION
**Pull request by czue**
Non-functional changes just getting the code around our current reindex workflows cleaned up and better tested. 

Probably easiest to review in the :ocean: with the :fish: :tropical_fish: 

@dimagi/scale-team cc @calellowitz @millerdev

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/tests/test_es_reindex.py [hunk 1]
```diff
@@ -22,3 +30,32 @@ def test_get_all_elastic_indices(self):
...
+        self.assertFalse(needs_reindex(self.elasticsearch, index_info))
+        ensure_index_deleted(index_info.index)
+
+
+def _make_test_index(es):
+    index_name = '{}{}'.format(TEST_ES_PREFIX, uuid.uuid4().hex)
+    alias_name = uuid.uuid4().hex
+    index_info = ElasticsearchIndexInfo(index=index_name, alias=alias_name)
+    create_index_and_set_settings_normal(es, index_name)
+    return index_info
```
Comment (benrudolph) [inline corehq/ex-submodules/pillowtop/tests/test_es_reindex.py]: :sunglasses:

File: corehq/ex-submodules/pillowtop/management/commands/ptop_es_manage.py [hunk 2]
```diff
@@ -74,8 +73,6 @@ def handle(self, *args, **options):
                 print 'Safety first!'
             return
 
-        if show_info:
```
Comment (benrudolph) [inline corehq/ex-submodules/pillowtop/management/commands/ptop_es_manage.py]: nit: can probably drop `show_info`

**Comment by benrudolph** [comment]
took a :swimmer: in the :ocean: and everything looked :bath:. :+1: on :white_check_mark:

File: corehq/ex-submodules/pillowtop/management/commands/ptop_es_manage.py (hunk 2 above)
Comment (czue) [inline corehq/ex-submodules/pillowtop/management/commands/ptop_es_manage.py]: good call. https://github.com/dimagi/commcare-hq/pull/10958/commits/c343943548dee5607a976e79b4984391d4207e73

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**